File: beepers/beepers.py
# python3 
# Arturas Aleksandrauskas artal938 - TDDD95
import sys
from itertools import permutations 

def manhatan(f, t):
    return abs(f[0] - t[0]) + abs(f[1] - t[1])

def calc_dist(start, pappers, min_d):
    tmp = start
    dist = 0
    for p in pappers:
        dist += manhatan(tmp, p)
        tmp = p
        if dist > min_d:
            return dist
    dist += manhatan(tmp, start)   
    return dist

def main():
    tests = int(sys.stdin.readline().strip())
    for _ in range(tests):
        maxX, maxY = map(int, sys.stdin.readline().split())
        startX, startY = map(int,sys.stdin.readline().split())
        pappers = []
        p = int(sys.stdin.readline().strip())
        for _ in range(p):
            x, y = map(int, sys.stdin.readline().split())
            pappers.append((x,y))
        dist = 400
        for perm in permutations(pappers):
            dist = min(dist, calc_dist((startX, startY), perm, dist))
        print(dist)
				
if __name__ == "__main__":
    main()

File: beepers/beepers.cpp
//Arturas Aleksandrauskas - artal938 TDDD95
#include <bits/stdc++.h>
using namespace std;

int manhatan(vector<pair<int, int>>::iterator f,vector<pair<int, int>>::iterator t){
    return abs(f->first-t->first) + abs(f->second-t->second);
}
int manhatan(pair<int, int> const & f,pair<int, int> const & t){
    return abs(f.first-t.first) + abs(f.second-t.second);
}

int calc_dist(pair<int, int> const & start, vector<pair<int,int>> & pappers, int min_d){
    auto tmp = pappers.begin();
    int dist{manhatan(*tmp, start)};
    for(auto p = pappers.begin()+1; p != pappers.end(); ++p){
        dist += manhatan(tmp, p);
        tmp = p;
        if (dist > min_d){
            return dist;
        }
    }
    dist += manhatan(*tmp, start);
    return dist;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    int p{}, t{}, startX{}, startY{}, x{}, y{}, dist{};
    cin >> t;
    vector<pair<int, int>> pappers;
    for (int i{}; i < t; i++){
        cin >> startX >> startY;
        cin >> startX >> startY;
        cin >> p;
        pappers.clear();
        pappers.reserve(p);
        for (int j{}; j < p; j++){
            cin >> x >> y;
            pappers.push_back(make_pair(x,y));
        }
        sort(pappers.begin(), pappers.end());
        dist = 400;
        do{
            dist = min(dist, calc_dist({startX, startY},pappers, dist));
        } while (next_permutation(pappers.begin(), pappers.end()));
        cout << dist << "\n";
    }

    
    return 0;
}
